surface/src/on_nurbs/fitting_curve_2d_asdm.cpp

```diff
@@ -289,7 +289,7 @@ FittingCurve2dASDM::assembleInterior (double wInt, double sigma2, double rScale,
     double param;
     Eigen::Vector2d pt, t, n;
     double error;
-    if (p < int (m_data->interior_param.size ()))
+    if (p < static_cast<unsigned int>(m_data->interior_param.size ()))
```

Comment:
`p` as `std::size_t` negates the need for cast, esp since other comparisons with `XYZ.size()` are uncasted